Change semgrep entry point, dispatch to osemgrep if --experimental

aryx opened the pull request:
test plan:
```
cd cli
pipenv install --dev
pipenv shell
time semgrep --help
0.335s
time semgrep --experimental --help
0.077s
```


PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

aryx committed:
Change semgrep entry point, dispatch to osemgrep if --experimental

test plan:
```
cd cli
pipenv install --dev
pipenv shell
time semgrep --help
0.335s
time semgrep --experimental --help
0.077s
```

aryx committed:
more

aryx committed:
more comments

aryx committed:
comments

aryx committed:
more comments

aryx committed:
fix osempass

aryx committed:
Merge branch 'develop' into dispatch_osemgrep

aryx committed:
changelog

tpetr committed:
temporary path fix for manylinux wheel test

apt-itude commented on cli/bin/semgrep:
```diff
@@ -1,6 +1,59 @@
+                print(f"semgrep-core is not a file")
+                # fatal error, see src/osemgrep/core/Exit_code.ml
+                sys.exit(2)
+    #TODO? Maybe better to just fallback to pysemgrep?
```
I think if you've explicitly asked for `--experimental` it makes sense to tell you the experiment failed

aryx replied:
true

apt-itude reviewed (approved):
Nice

tpetr committed:
tweak

tpetr committed:
thats what I get for trying to be clever

aryx committed:
fix for Homebrew

test plan:
```
$ pipenv install --dev
$ pipenv shell
$ semgrep --help
...
$ rm ~/semgrep/cli/src/semgrep/bin/semgrep
$ semgrep --help # still work, use semgrep-core in my path
...
$ rm ~/.opam/4.14.0/.../semgrep-core
$ semgrep -help
Failed to find semgrep-core (needed for --experimental) in PATH or in the semgrep package
```
